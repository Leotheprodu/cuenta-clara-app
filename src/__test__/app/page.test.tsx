import Home from "@/app/page";
import { render, screen, within } from "@testing-library/react";
import { expect, test } from "vitest";

test("render home page", () => {
  render(<Home />);
  const main = within(screen.getByRole("main"));
  expect(
    main.getByRole("heading", {
      level: 1,
      name: "Tu negocio siempre a mano",
    })
  ).toBeDefined();
});

test("render home page to same of the snapshot", () => {
  const { container } = render(<Home />);
  expect(container).toMatchSnapshot();
});
