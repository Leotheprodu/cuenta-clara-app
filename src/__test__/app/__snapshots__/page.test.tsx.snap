// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render home page to same of the snapshot 1`] = `
<div>
  <main
    class=""
  >
    <div
      class="bg-primario"
    >
      <section
        class="bg-primario w-full flex flex-col sm:flex-row top-12 pt-[5rem] h-screen overflow-hidden"
        style="opacity: 1;"
      >
        <div
          class="px-2 sm:w-1/3 flex flex-col items-center justify-center gap-10"
        >
          <h1
            class="text-blanco text-6xl text-center p-1 sm:p-5 font-semibold"
          >
            Tu 
            <span
              class="gradient-text"
            >
              negocio
            </span>
             siempre a mano
          </h1>
          <h2
            class="text-blanco/75 text-xl text-center p-1 sm:p-5"
          >
            <span
              class="gradient-text uppercase"
            >
              Yehu
            </span>
             mejora la experiencia de tus clientes, permitiendo consultar sus facturas y realizar pagos
          </h2>
        </div>
        <div
          class="sm:w-2/3 relative"
          style="transform: translateX(0px) translateX(2000px) translateZ(0);"
        >
          <svg
            class="absolute top-0 scale-105"
            viewBox="0 0 1440 320"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0,192L48,181.3C96,171,192,149,288,128C384,107,480,85,576,112C672,139,768,213,864,229.3C960,245,1056,203,1152,170.7C1248,139,1344,117,1392,106.7L1440,96L1440,0L1392,0C1344,0,1248,0,1152,0C1056,0,960,0,864,0C768,0,672,0,576,0C480,0,384,0,288,0C192,0,96,0,48,0L0,0Z"
              fill="#001d3d"
              fill-opacity="1"
            />
          </svg>
          <img
            alt="vista en computadora"
            class=""
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="800"
            src="/_next/image?url=%2Fdesk1.png&w=3840&q=75"
            srcset="/_next/image?url=%2Fdesk1.png&w=1200&q=75 1x, /_next/image?url=%2Fdesk1.png&w=3840&q=75 2x"
            style="color: transparent;"
            width="1200"
          />
          <svg
            class="absolute bottom-0 rotate-180 scale-105"
            viewBox="0 0 1440 320"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0,192L48,181.3C96,171,192,149,288,128C384,107,480,85,576,112C672,139,768,213,864,229.3C960,245,1056,203,1152,170.7C1248,139,1344,117,1392,106.7L1440,96L1440,0L1392,0C1344,0,1248,0,1152,0C1056,0,960,0,864,0C768,0,672,0,576,0C480,0,384,0,288,0C192,0,96,0,48,0L0,0Z"
              fill="#001d3d"
              fill-opacity="1"
            />
          </svg>
        </div>
      </section>
      <section
        class="bg-terciario overflow-hidden sm:h-[300vh] relative"
      >
        <div
          class="flex flex-col w-full sm:flex-row sm:fixed top-10 sm:top-1/4"
          style="opacity: 0; transform: translateY(1000px) translateZ(0);"
        >
          <div
            class="sm:w-1/2 flex items-center sm:items-start justify-center sm:justify-start"
          >
            <img
              alt="vista en celular"
              class=""
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="800"
              src="/_next/image?url=%2Fphone1.png&w=3840&q=75"
              srcset="/_next/image?url=%2Fphone1.png&w=1200&q=75 1x, /_next/image?url=%2Fphone1.png&w=3840&q=75 2x"
              style="color: transparent;"
              width="1200"
            />
          </div>
          <div
            class="px-2sm:w-1/2 flex flex-col gap-10"
          >
            <h1
              class="text-blanco text-center sm:text-left text-6xl p-1 sm:p-5 font-semibold"
            >
              Con 
              <span
                class="text-secundario uppercase"
              >
                Yehu
              </span>
               
              Puedes
            </h1>
            <ul
              class="text-secundario text-2xl text-center sm:text-left p-1 sm:p-5 flex flex-col justify-center gap-3 "
            >
              <li
                class="bg-gradient-to-r p-1 rounded-md from-cuaternario to-terciario w-full"
              >
                Registrar y contactar rapidamente tus clientes
              </li>
              <li
                class="bg-gradient-to-r p-1 rounded-md from-cuaternario to-terciario w-full"
              >
                Controlar tu inventario
              </li>
              <li
                class="bg-gradient-to-r p-1 rounded-md from-cuaternario to-terciario w-full"
              >
                Registrar facturas y transacciones de tus clientes
              </li>
              <li
                class="bg-gradient-to-r p-1 rounded-md from-cuaternario to-terciario w-full"
              >
                Controlar los saldos de tus clientes
              </li>
              <li
                class="bg-gradient-to-r p-1 rounded-md from-cuaternario to-terciario w-full"
              >
                Consultar tus ventas
              </li>
            </ul>
          </div>
        </div>
      </section>
    </div>
  </main>
</div>
`;
